File: SQL/Select/SQLSelectField.vb

' ___________________________________________________
'
'  © Hi-Integrity Systems 2010. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace SQL

    Public Class SQLSelectField

        Private pstrAlias As String
        Private pobjExpression As SQLExpression

        Public Sub New(ByVal objExpression As SQLExpression)

            If objExpression Is Nothing Then
                Throw New ArgumentNullException
            End If

            pobjExpression = objExpression

        End Sub

        Friend ReadOnly Property Expression As SQLExpression
            Get

                Return pobjExpression

            End Get
        End Property

        Public Property [Alias]() As String
            Get

                [Alias] = pstrAlias

            End Get

            Set(ByVal Value As String)

                pstrAlias = Value.Trim

            End Set
        End Property

        Friend Function SQL(ByVal eConnectionType As Database.ConnectionType) As String

            Dim strSQL As String = pobjExpression.SQL(eConnectionType)

            If Me.Alias <> String.Empty Then
                strSQL &= " AS " & SQLConvertIdentifierName(Me.Alias, eConnectionType)
            End If

            Return strSQL

        End Function

    End Class

End Namespace

File: SQL/Select/SQLSelectField.cs
// _________________________________________________________________________
//
//  © Hi-Integrity Systems 2010. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// 
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//	    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// _________________________________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLSelectField
	{
		private string pstrAlias;
		private SQLExpression pobjExpression;
			
		public SQLSelectField(SQLExpression objExpression)
		{
			if (objExpression == null)
				throw new ArgumentNullException();
				
			pobjExpression = objExpression;
		}
			
		internal SQLExpression Expression
		{
			get
			{
				return pobjExpression;
			}
		}
			
		public string Alias
		{
			get
			{
                return pstrAlias;
			}
				
			set
			{
				pstrAlias = value;
			}
		}
	}
}
